wallet/src/main/scala/org/bitcoins/wallet/api/WalletApi.scala

```diff
@@ -33,15 +23,13 @@ import scala.concurrent.Future
   */
 sealed trait WalletApi {
 
-  def dbConfig: DbConfig
+  def walletAppConfig: WalletAppConfig
 
-  def chainParams: ChainParams
+  def dbConfig: WalletDbConfig = walletAppConfig.dbConfig
 
-  def networkParameters: BitcoinNetwork = chainParams match {
-    case MainNetChainParams    => MainNet
-    case TestNetChainParams    => TestNet3
-    case RegTestNetChainParams => RegTest
-  }
+  def chainParams: ChainParams = walletAppConfig.chain
+
+  def networkParameters: NetworkParameters = walletAppConfig.network
```

Comment:
```suggestion
def networkParameters: BitcoinNetwork = walletAppConfig.network
```

That way we avoid the cast in Wallet.scala

Reply:
I actually want to try and keep this generic for when we support multiple cryptocurrency networks, it is important to keep the `Api` level stuff agnostic to the cryptocurrency network

Reply:
I took the easy way out by casting for now, I should really go fix the cast and pattern match